packages/remirror__extension-link/src/link-extension.ts

```diff
@@ -683,6 +465,94 @@ export class LinkExtension extends MarkExtension<LinkOptions> {
             this.options.onUpdateLink(doc.textBetween(from, to), meta);
           }
         });
+
+        if (!this.options.autoLink) {
+          return;
+        }
+
+        const docChanged = transactions.some((tr) => tr.docChanged);
+
+        if (!docChanged) {
+          return;
+        }
+
+        // Create a single transaction, by combining all transactions
+        const composedTransaction = prevState.tr;
+        transactions.forEach((transaction) => {
+          transaction.steps.forEach((step) => {
+            composedTransaction.step(step);
+          });
+        });
+
+        const changes = getChangedRanges(composedTransaction);
+        const { mapping } = composedTransaction;
+        const { tr, doc } = state;
+
+        const { updateLink, removeLink } = this.store.chain(tr);
+
+        changes.forEach(({ from, to, prevFrom, prevTo }) => {
+          // Remove auto links that are no longer valid
+          this.getLinkMarksInRange(prevState.doc, prevFrom, prevTo, true).forEach((prevMark) => {
+            const newFrom = mapping.map(prevMark.from);
+            const newTo = mapping.map(prevMark.to);
+            const newMarks = this.getLinkMarksInRange(doc, newFrom, newTo, true);
+
+            newMarks.forEach((newMark) => {
+              const wasLink = this._autoLinkRegexNonGlobal?.test(prevMark.text);
+              const isLink = this._autoLinkRegexNonGlobal?.test(newMark.text);
+
+              if (wasLink && !isLink) {
+                removeLink({ from: newMark.from, to: newMark.to }).tr();
+              }
+            });
+          });
+
+          // Store all the callbacks we need to make
+          const onUpdateCallbacks: Array<Pick<EventMeta, 'range' | 'attrs'> & { text: string }> =
+            [];
+
+          // Find text that can be auto linked
+          tr.doc.nodesBetween(from, to, (node, pos) => {
+            if (!node.isTextblock) {
+              return;
+            }
+
+            const nodeText = tr.doc.textBetween(pos, pos + node.nodeSize, undefined, ' ');
+            this.findAutoLinks(nodeText)
+              // calculate link position
+              .map((link) => ({
+                ...link,
+                from: pos + link.start + 1,
+                to: pos + link.end + 1,
+              }))
+              .filter((link) => {
+                // Determine if found link is within the changed range
+                return within(link.from, from, to) || within(link.to, from, to);
```

Comment:
@whawker  Seems that "editing inside a link won't update href" issue comes from here